scanpipe/pipes/resolve.py

```diff
@@ -59,6 +62,16 @@ def resolve_packages(input_location):
     return resolved_packages
 
 
+def get_manifest_resources(project):
+    """Get a list of locations for all package manifests in a project"""
+    for resource in project.codebaseresources.no_status():
+        manifest_type = get_default_package_type(input_location=resource.location)
+        if manifest_type:
+            resource.update(status=flag.APPLICATION_PACKAGE)
+
+    return project.codebaseresources.filter(status=flag.APPLICATION_PACKAGE)
```

Comment:
I'm not sure how to optimize this further, could you suggest how to do this here?

> we do 1 query per detected manifest_type

Yes we are updating the status for each manifest present, and we can't optimize this to get all the manifests at once because we have to apply `handler.is_datafile(location)` on every location individually.

We can create a filter here by looping through each handler type and collecting ``path_patterns`` for each handler type, and then create a queryset by filtering by these patterns, but this would not be correct. Because the default ``is_datafile`` implementation here https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/models.py#L974 is more than just path patterns, and this is also often overridden by ecosystem classes that inherit from it.

We also cannot collect all the resources in a list and then create a queryset to update it at once, as this would not be ideal? I couldn't find examples for this elsewhere as everywhere we have to update each resource/object individually we run update() individually.